default/decorators.py

```diff
@@ -41,3 +41,42 @@ def wrap(request, *args, **kwargs):
         return function(request, *args, **kwargs)
 
     return wrap
+
+
+def validate_split_size(function):
+    @wraps(function)
+    def wrap(request, *args, **kwargs):
+        split_size = request.GET.get('splitSize')
+        if split_size.isdecimal():
+            kwargs['size'] = int(split_size)
+        else:
+            msg = _('An invalid split size was submitted, and therefore ignored. Default value is used, split size: 1')
+            if 'warnings' in kwargs:
+                kwargs['warnings'].append(msg)
+            else:
+                kwargs['warnings'] = [msg]
+        return function(request, *args, **kwargs)
+
+    return wrap
+
+
+def optional_args(function):
```

Comment:
Can we change the name of this function to `validate_encoding`? I understand that this is what is done here.

Reply:
Validate `pretty_json` too, changed to `validate_optional_args`.